service/pricing/cid_gravity.go

```diff
@@ -0,0 +1,185 @@
+package pricing
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	pb "github.com/textileio/bidbot/gen/v1"
+	golog "github.com/textileio/go-log/v2"
+)
+
+const (
+	// If loading rules from CID gravity API takes longer than this timeout, turn it into background.
+	cidGravityLoadRulesTimeout = 5 * time.Second
+)
+
+var (
+	log              = golog.Logger("bidbot/pricing")
+	cidGravityAPIUrl = "https://api.cidgravity.com/api/integrations/bidbot"
+	// Use cached rules if they are loaded no earlier than this period.
+	cidGravityCachePeriod = time.Minute
+)
+
+// CIDGravityRules is the format CID gravity API returns. Needs to be public to unmarshal JSON payload. Do not use.
+type CIDGravityRules struct {
+	Blocked         bool
+	MaintenanceMode bool
+	PricingRules    []struct {
+		Verified    bool
+		MinSize     uint64 // bytes
+		MaxSize     uint64 // bytes
+		MinDuration uint64 // epoches
+		MaxDuration uint64 // epoches
+		Price       int64  // attoFil
+	}
+}
+
+type cidGravityRules struct {
+	apiKey      string
+	clientRules map[string]*clientRules
+	lkRules     sync.Mutex
+}
+
+// NewCIDGravityRules returns PricingRules based on CID gravity configuration for the storage provider.
+func NewCIDGravityRules(apiKey string) PricingRules {
```

Comment:
Maybe not a crash, but probably the error will be discovered after boot when a bid happen.. and that's a bit late and annoying compared to "failing as fast as it boots"?

Anyway, SGTM since I know there's an `if` check from the caller; just as a second layer security. 
Ignore my pranoid :)